### ISSUE TITLE
Limit profile publishing based on hunt overlap

### ISSUE THREAD

**Issue by ebroder** [deathandmayhem/jolly-roger#629]
Currently, profiles are published to all logged in users. If we allow people to start creating their own hunts, we need to contemplate the possibility that someone wouldn't want their data (which includes sensitive stuff like phone number) shared with other users they have no connection to.

This should be moderately straightforward once #628 is complete, since we can add a `{ hunts: { $in: user.hunts } }` term to the query against users (because of how Mongo handles queries against arrays, the `$in` will match against both the array in its entirety and - more importantly for our purposes - each of the individual array entries).

**Comment by zarvox** [comment]
We should also think about the lifecycle around joining a hunt -- today, when you're invited to a hunt, the hunt gets added to your user object immediately, without you taking any action, but if sharing a hunt with someone means revealing profile information, then an attacker that knows a target's email address can gain access to the rest of their profile by inviting them to a hunt, and there's no way to leave a hunt.

I could imagine making hunt member a two-step invitation flow, where both the inviter and the invitee must indicate their desire for the invitee to join a hunt before it happens?  That would add some friction, but be more resilient to the described attack.

**Comment by ebroder** [comment]
Yeah good point. I'll split out a new ticket for that, because I think that can be done as a separate refactor. (#635)
